validation/citrus-validation-json/src/main/java/org/citrusframework/validation/json/JsonTextMessageValidator.java

```diff
@@ -56,225 +46,85 @@
-                validateJson("$.", (JSONObject) receivedJson, (JSONObject) controlJson, validationContext, context, readContext);
-            } else if (receivedJson instanceof JSONArray) {
-                JSONObject tempReceived = new JSONObject();
-                tempReceived.put("array", receivedJson);
-                JSONObject tempControl = new JSONObject();
-                tempControl.put("array", controlJson);
-
-                validateJson("$.", tempReceived, tempControl, validationContext, context, readContext);
-            } else {
-                throw new CitrusRuntimeException("Unsupported json type " + receivedJson.getClass());
-            }
-        } catch (IllegalArgumentException e) {
-            throw new ValidationException(String.format("Failed to validate JSON text:%n%s", receivedJsonText), e);
-        } catch (ParseException e) {
-            throw new CitrusRuntimeException("Failed to parse JSON text", e);
+        if (!hasText(controlJsonText)) {
+            logger.debug("Skip message payload validation as no control message was defined");
+            return;
+        } else if (!hasText(receivedJsonText)) {
+            throw new ValidationException("Validation failed - expected message contents, but received empty message!");
         }
 
+        new JsonElementValidator(strict, context, validationContext.getIgnoreExpressions()).validate(
+                parseJson(permissiveMode, receivedJsonText, controlJsonText)
+        );
         logger.info("JSON message validation successful: All values OK");
     }
 
-    /**
-     * Find json schema repositories in test context.
-     * @param context
-     * @return
-     */
-    private List<JsonSchemaRepository> findSchemaRepositories(TestContext context) {
-        return new ArrayList<>(context.getReferenceResolver().resolveAll(JsonSchemaRepository.class).values());
-    }
-
     /**
      * Validates JSON text with comparison to expected control JSON object.
      * JSON entries can be ignored with ignore placeholder.
+     * <p>
+     * @deprecated use {@link JsonElementValidator#validate} instead
      *
-     * @param elementName the current element name that is under verification in this method
-     * @param receivedJson the received JSON text object.
-     * @param controlJson the expected control JSON text.
+     * @param elementName       the current element name that is under verification in this method
+     * @param receivedJson      the received JSON text object.
+     * @param controlJson       the expected control JSON text.
      * @param validationContext the JSON message validation context.
-     * @param context the current test context.
-     * @param readContext the JSONPath read context.
+     * @param context           the current test context.
+     * @param readContext       the JSONPath read context.
      */
-    @SuppressWarnings("rawtypes")
+    @Deprecated(forRemoval = true)
```

Comment:
If we deprecate we should call the method as part of the new logic. Otherwise when the method is not called I would rather remove and break the API. Then users that have added customizations to this method get noticed when updating and can adjust accordingly.

@bbortt WDYT?

Reply:
that argumentation is valid. it support it 😉